x-hooks/use-bounty.tsx

```diff
@@ -0,0 +1,203 @@
+    if (!state.Service?.active || !state.Service?.network || !state.currentBounty?.data?.contractId || state.spinners?.bountyChain)
+      return;
+
+    if (!force && isCurrentBountyCached())
+      return;
+
+    dispatch(changeSpinners.update({bountyChain: true}))
+
+    state.Service.active.getBounty(state.currentBounty.data.contractId)
+      .then(bounty => {
+
+        const pullRequestsMapper = (pullRequest) => ({
+          ...pullRequest,
+          isCancelable: !bounty.proposals.find(proposal => proposal.prId === pullRequest.id)
+        });
+
+        bounty.pullRequests = bounty.pullRequests.filter(pr => !pr.canceled).map(pullRequestsMapper);
+        bounty.fundedAmount = bounty.funding.reduce((p, c) => p.plus(c.amount), BigNumber(0))
+        bounty.fundedPercent = bounty.fundedAmount.multipliedBy(100).dividedBy(bounty.fundingAmount);
+        bounty.isFundingRequest = bounty.fundingAmount.gt(0);
+
+        // todo: missing tokenInformation
+
+        dispatch(changeCurrentBountyDataChain.update(bounty));
+
+        state.Service.active.isBountyInDraftChain(bounty.creationDate)
+          .then(bool => dispatch(changeCurrentBountyDataIsDraft(bool)));
+
+        bountyReadyPRsHasNoInvalidProposals(bounty, state.Service.network)
+          .catch(() => -1)
+          .then(value => {
+            dispatch(changeCurrentBountyDataIsFinished(value !== 0));
+            dispatch(changeCurrentBountyDataIsInValidation([2, 3].includes(value)))
+            dispatch(changeSpinners.update({bountyChain: false}))
+          });
+
+      });
+
+  }
+
+  /**
+   *  todo: getExtendedProposalsForCurrentBounty() should happen on webnetwork-events
+   *
+   *  MAKE SURE that these functions (getExtendedProposalsForCurrentBounty, getExtendedPullRequestsForCurrentBounty)
+   *  are only called once, since they ignore cached information
+   */
+
+  function getExtendedProposalsForCurrentBounty() {
+    if (!state.currentBounty?.chainData || !state.Service?.active)
+      return Promise.resolve([]);
+
+    const bounty = state.currentBounty.chainData;
+    const dbBounty = state.currentBounty.data;
+    const wallet = state.currentUser?.walletAddress;
+
+    return Promise.all(bounty.proposals.map(proposal =>
+        (dbBounty.merged
+          ? Promise.resolve(+dbBounty.merged != proposal.id)
+          : state.Service.active.isProposalDisputed(+bounty.id, proposal.id)
+        ).then(isDisputed =>
+          !wallet
+            ? ({...proposal, isDisputed})
+            : state.Service.active.getDisputesOf(wallet, +bounty.id, +proposal.id)
+              .then(value => ({...proposal, isDisputed, canUserDispute: !value})))))
+      .then(proposals => {
+        // dispatch(changeCurrentBountyDataProposals(proposals));
+        return proposals;
+      })
+      .catch(e => {
+        console.error(`Failed to get extended proposals`, e);
+        return;
+      });
+  }
+
+  function getExtendedPullRequestsForCurrentBounty() {
```

Comment:
Could `getExtendedProposalsForCurrentBounty` and `getExtendedPullRequestsForCurrentBounty`  be used in `getChainBounty` and `getDatabaseBounty` ?

Reply:
@vhcsilva as we saw yesterday, if we do that we set an expensive behaviour on the application; Since these two functions should only be needed _some times_ and not _everytime_ we want a bounty data -- these need to be called by hand, so we make sporadic use of them _and_ control when they are called so we don't end up on making this call too many times.